x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/AbstractScriptFieldScript.java

```diff
@@ -93,5 +101,19 @@ public final void setDocument(int docId) {
         return leafSearchLookup.doc();
     }
 
+    protected final void checkMaxSize(int currentSize) {
```

Comment:
is there a chance that we have some standard emitValue in the base class that calls this method? Otherwise all impls should call it but now only string does?

Reply:
I don't think we can share a base class implementation, not without auto-boxing.

Reply:
ok, but scripts other than strings should also check for max size? or do they already do so and I have missed it?

Reply:
I'd thought I was just going to do strings in the first PR and grab the others in a follow up one. But, yeah, they should all call this method. Well, `boolean`s may not have to because we don't actually allocate anything when they emit values. They are probably safe being limited by painless's standard stuff.

Reply:
it seems easy enough to do them all in this PR, what do you think?

Reply:
I can give it a shot, sure!

Reply:
I've pushed!